contracts/src/messaging/MessageBus.sol

```diff
@@ -79,9 +89,10 @@ contract MessageBus is IMessageBus, Initializable, OwnableUpgradeable {
         uint32 topic,
         bytes calldata payload,
         uint8 consistencyLevel
-    ) external override returns (uint64 sequence) {
-        //TODO: implement messageFee mechanism.
-        //require(msg.value >= messageFee());
+    ) external payable override returns (uint64 sequence) {
+        if (address(fees) != address(0)) { // No fee required for L1 to L2 messages.
+            require(msg.value >= getMessageFee(payload.length), "Insufficient funds to publish message");
```

Comment:
I don't see where the value is transferred to the contract

Reply:
Here we check that the message bus has received enough. I guess it would be good to transfer it to the fees contract tho for better accounting, which in turn will transfer out to some EOA when owner requests